### ISSUE TITLE
Any way to use pushing with an external API?

### ISSUE THREAD

**Issue by faelsoto** [yuki24/pushing#19]
Hey, great job! I think something like this should be included by default on Rails.

We currently have a basic wrapper for push notifications, but testing it will be a mess. I found pushing searching for actionmailer push notifications, since actionmailer stores internally the mailers it has sent, it is possible to `assert_difference` within it.

However we're using it with OneSignal. To be honest I hadn't heard about FCM/APN until today, and what I can grasp about it, it seems like too much work. So why not outsourcing it?

The question is if it is possible (or it is within the scope) to configure and use an external API?

**Comment by yuki24** [comment]
Absolutely! Pushing is designed to be able to send request to multiple [platforms](https://github.com/yuki24/pushing/blob/master/lib/pushing/platforms.rb) using [the adapter pattern](https://github.com/yuki24/pushing/blob/master/lib/pushing/adapters.rb#L7-L12). Theoretically it should be possible to add support for other services OneSignal or even [Amazon SNS](https://docs.aws.amazon.com/sns/latest/dg/SNSMobilePush.html). I'm not familiar with OneSignal though. It would be greatly appreciated if you could take a stab at it.
